conf/layer.conf

```diff
@@ -18,24 +18,20 @@ AOS_UNIT_MODEL ?= "aos-unit"
 AOS_UNIT_VERSION ?= "1.0"
 
 AOS_NODE_ID ?= "node0"
-AOS_NODE_TYPE ?= "${AOS_UNIT_MODEL}-${AOS_UNIT_VERSION}-${AOS_NODE_ID}"
 AOS_MAIN_NODE_ID ?= "${AOS_NODE_ID}"
 
-AOS_RUNNER ??= "crun"
+AOS_NODE_TYPE ?= "${AOS_UNIT_MODEL}-${AOS_UNIT_VERSION}-${AOS_NODE_ID}"
 
-AOS_SM_NODES ?= "${AOS_NODE_ID}"
-AOS_UM_NODES ?= "${AOS_NODE_ID}"
-AOS_IAM_NODES ?= "${AOS_NODE_ID}"
+AOS_RUNNER ??= "crun"
 
 AOS_NODE_HOSTNAME ?= "${AOS_NODE_ID}"
 AOS_MAIN_NODE_HOSTNAME ?= "${AOS_MAIN_NODE_ID}"
-AOS_IAM_HOSTNAMES ?= "${AOS_IAM_NODES}"
 
 AOS_UM_COMPONENT_PREFIX ?= ""
 
 # Overrides
 
-OVERRIDES:append = "${@':aos-main-node' if d.getVar('AOS_NODE_ID') == d.getVar('AOS_MAIN_NODE_ID') else ':aos-secondary-node'}"
+OVERRIDES:append = "${@':aos-main-node' if d.getVar('AOS_NODE_HOSTNAME') == d.getVar('AOS_MAIN_NODE_HOSTNAME') else ':aos-secondary-node'}"
```

Comment:
Let's have dedicated variable AOS_MAIN_NODE that explicitly defines main node.

Reply:
fixed